java-frontend/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -923,7 +923,7 @@ private void buildBreakStatement(BreakStatementTree tree) {
   }
 
   private void buildYieldStatement(YieldStatementTree tree) {
-    currentBlock = createUnconditionalJump(tree, breakTargets.getLast(), currentBlock);
+    currentBlock = createUnconditionalJump(tree, breakTargets.isEmpty() ? null : breakTargets.getLast(), currentBlock);
```

Comment:
Finding the right place to test the CFG is a bit complicated but I think a good candidate would be the tests for `MethodTreeImpl` that has a CFG method. With the new fix, we should not fail to generate a CFG for the example method that you added in `java-checks/src/test/files/checks/ParameterReassignedToCheck.java` . In particular, the [compute_cfg test](https://github.com/SonarSource/sonar-java/blob/71f5cb1008da6e42e70412012ec95f3898776209/java-frontend/src/test/java/org/sonar/java/model/declaration/MethodTreeImplTest.java#L198) looks like a good place to add test cases.